crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -977,51 +870,74 @@ pub mod world {
     #[generate_trait]
     impl Self of SelfTrait {
         #[inline(always)]
-        fn get_account_address(self: @ContractState) -> ContractAddress {
-            get_tx_info().unbox().account_contract_address
+        fn is_caller_world_owner(self: @ContractState) -> bool {
+            self.is_owner(WORLD, get_caller_address())
         }
 
-        /// Verifies if the calling account is owner of the resource or the
-        /// owner of the world.
+        /// To be sure that a world function has been directly called from an account (with sozo for
+        /// example), to avoid any malicious contract between the account and the world to be able
+        /// to call some functions with account privileges.
+        #[inline(always)]
+        fn must_be_a_direct_call_from_account(self: @ContractState) {
+            let caller = get_caller_address();
+            let account = get_tx_info().unbox().account_contract_address;
+
+            assert(caller == account, Errors::NOT_A_DIRECT_CALL);
+        }
+
+        /// Panic if the caller is NOT an owner of the resource.
         ///
         /// # Arguments
-        ///
-        /// * `resource` - The selector of the resource being verified.
-        ///
-        /// # Returns
-        ///
-        /// * `bool` - True if the calling account is the owner of the resource or the owner of the
-        /// world,
-        ///            false otherwise.
+        ///   * `resource_id` - the id of the resource.
```

Comment:
We should start to homogenize the term `resource selector` instead of using `id`.

`id` is mostly used at database level in the core. Otherwise we have `selector` at higher level for the world.

Reply:
Yes, I agree 👍